ember-app/app/visitors/issue/references.js

```diff
@@ -0,0 +1,54 @@
+import Ember from 'ember';
+
+var IssueReferencesVisitor = Ember.Object.create({
+  visit: function(issue){
+    var promises = [
+      this.run(this.discoverIssues, issue),
+      this.run(this.discoverClosedIssues, issue)
+    ];
+
+    Ember.RSVP.all(promises).then((references)=> {
+      issue.set('issueReferences', _.flatten(references));
+    });
+  },
+
+  run: function(discovery, issue){
+    var _self = this;
+    return new Ember.RSVP.Promise((resolve, reject)=>{
+      resolve(discovery.call(_self, issue));
+      reject([]);
+    });
+  },
+
+  discoverIssues: function(issue){
+    var references = issue.get('cardRelationships')['issue-references'] || [];
+    var issuesById = issue.get('repo.board.issuesById');
+    return references.map((reference)=>{
+      if(issuesById.hasOwnProperty(reference.id)){
+        return issuesById[reference.id][0];
+      }
+    }).compact();
+  },
+  discoverClosedIssues: function(issue){
+    var references = issue.get('cardRelationships')['issue-references'] || [];
+    var discovered_issues = this.discoverIssues(issue);
```

Comment:
I guess it depends how we ultimately want to handle missing issues, closed or otherwise (Create 'best guess' data or actually fetch the issue?) 

`return references.map(ref => discoverOpenIssue(ref) || discoverClosedIssue(ref) || discoverMissingIssue(ref));` 

Would work synchronously, but would potentially invoke 3 promises per reference.